channel/types/channel_state.go

```diff
@@ -0,0 +1,29 @@
+package types
+
+import (
+	"perun.network/go-perun/channel"
+	"reflect"
+)
+
+type ChannelState struct {
+	ID       channel.ID
+	Balances []uint64
+	Version  uint64
+	Final    bool
+}
+
+func MakeChannelState(id channel.ID, balances []uint64, version uint64, final bool) ChannelState {
+	return ChannelState{
+		ID:       id,
+		Balances: balances,
+		Version:  version,
+		Final:    final,
+	}
+}
+
+func (cs ChannelState) Equal(other ChannelState) bool {
+	return cs.ID == other.ID &&
+		reflect.DeepEqual(cs.Balances, other.Balances) &&
```

Comment:
`reflect` is a package we generally avoid. Implementing a simple `Equal` method between two balances is easy enough by iterating over both slices after checking their equal range.